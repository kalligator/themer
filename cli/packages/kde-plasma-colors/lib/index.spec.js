const { render, renderInstructions } = require('./index');
const { colors } = require('../../colors-default');

describe('themer kde-plasma-colors theme generator', () => {
  const promisedFiles = Promise.all(render(colors));
  it('renders valid theme files', async () => {
    const files = await promisedFiles;
    files.forEach((file) => {
      expect(file.contents.toString('utf8')).toMatchSnapshot();
    });
  });
  it('renders instructions', async () => {
    const files = await promisedFiles;
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();
  });
});
