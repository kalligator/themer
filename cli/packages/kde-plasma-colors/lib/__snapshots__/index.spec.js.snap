// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themer kde-plasma-colors theme generator renders instructions 1`] = `
"
1. Open System Settings
2. Navigate to Appearance > Colors
3. Click \\"Install from File...\\" and choose the generated theme file (\`ThemerDark.colors\` or \`ThemerLight.colors\`)
"
`;

exports[`themer kde-plasma-colors theme generator renders valid theme files 1`] = `
"[ColorEffects:Disabled]
Color=101,96,102
ColorAmount=0
ColorEffect=0
ContrastAmount=0.65
ContrastEffect=1
IntensityAmount=0.1
IntensityEffect=2

[ColorEffects:Inactive]
ChangeSelectionColor=true
Color=132,126,133
ColorAmount=0.025
ColorEffect=2
ContrastAmount=0.1
ContrastEffect=2
Enable=false
IntensityAmount=0
IntensityEffect=0

[Colors:Button]
BackgroundAlternate=101,96,102
BackgroundNormal=40,38,41
DecorationFocus=38,201,158
DecorationHover=38,201,158
ForegroundActive=38,201,158
ForegroundInactive=162,157,163
ForegroundLink=102,191,255
ForegroundNegative=255,64,80
ForegroundNeutral=242,129,68
ForegroundNormal=255,252,255
ForegroundPositive=164,204,53
ForegroundVisited=204,120,250

[Colors:Complementary]
BackgroundAlternate=101,96,102
BackgroundNormal=40,38,41
DecorationFocus=38,201,158
DecorationHover=38,201,158
ForegroundActive=38,201,158
ForegroundInactive=162,157,163
ForegroundLink=102,191,255
ForegroundNegative=255,64,80
ForegroundNeutral=242,129,68
ForegroundNormal=255,252,255
ForegroundPositive=164,204,53
ForegroundVisited=162,157,163

[Colors:Header]
BackgroundAlternate=40,38,41
BackgroundNormal=40,38,41
DecorationFocus=38,201,158
DecorationHover=38,201,158
ForegroundActive=38,201,158
ForegroundInactive=162,157,163
ForegroundLink=102,191,255
ForegroundNegative=255,64,80
ForegroundNeutral=242,129,68
ForegroundNormal=255,252,255
ForegroundPositive=164,204,53
ForegroundVisited=204,120,250

[Colors:Header][Inactive]
BackgroundAlternate=40,38,41
BackgroundNormal=40,38,41
DecorationFocus=38,201,158
DecorationHover=38,201,158
ForegroundActive=38,201,158
ForegroundInactive=162,157,163
ForegroundLink=102,191,255
ForegroundNegative=255,64,80
ForegroundNeutral=242,129,68
ForegroundNormal=255,252,255
ForegroundPositive=164,204,53
ForegroundVisited=204,120,250

[Colors:Selection]
BackgroundAlternate=224,220,224
BackgroundNormal=255,210,74
DecorationFocus=38,201,158
DecorationHover=38,201,158
ForegroundActive=40,38,41
ForegroundInactive=101,96,102
ForegroundLink=201.25,167,65.75
ForegroundNegative=201.25,57.5,70.25
ForegroundNeutral=191.5,106.25,61.25
ForegroundNormal=40,38,41
ForegroundPositive=133,162.5,50
ForegroundVisited=163,99.5,197.75

[Colors:Tooltip]
BackgroundAlternate=40,38,41
BackgroundNormal=40,38,41
DecorationFocus=38,201,158
DecorationHover=38,201,158
ForegroundActive=38,201,158
ForegroundInactive=162,157,163
ForegroundLink=102,191,255
ForegroundNegative=255,64,80
ForegroundNeutral=242,129,68
ForegroundNormal=255,252,255
ForegroundPositive=164,204,53
ForegroundVisited=204,120,250

[Colors:View]
BackgroundAlternate=40,38,41
BackgroundNormal=40,38,41
DecorationFocus=38,201,158
DecorationHover=38,201,158
ForegroundActive=38,201,158
ForegroundInactive=162,157,163
ForegroundLink=102,191,255
ForegroundNegative=255,64,80
ForegroundNeutral=242,129,68
ForegroundNormal=255,252,255
ForegroundPositive=164,204,53
ForegroundVisited=204,120,250

[Colors:Window]
BackgroundAlternate=40,38,41
BackgroundNormal=40,38,41
DecorationFocus=38,201,158
DecorationHover=38,201,158
ForegroundActive=38,201,158
ForegroundInactive=162,157,163
ForegroundLink=102,191,255
ForegroundNegative=255,64,80
ForegroundNeutral=242,129,68
ForegroundNormal=255,252,255
ForegroundPositive=164,204,53
ForegroundVisited=204,120,250

[General]
ColorScheme=ThemerDark
Name=Themer Dark
shadeSortColumn=true

[KDE]
contrast=4

[WM]
activeBackground=40,38,41
activeBlend=255,252,255
activeForeground=255,252,255
inactiveBackground=40,38,41
inactiveBlend=162,157,163
inactiveForeground=162,157,163"
`;

exports[`themer kde-plasma-colors theme generator renders valid theme files 2`] = `
"[ColorEffects:Disabled]
Color=193,188,194
ColorAmount=0
ColorEffect=0
ContrastAmount=0.65
ContrastEffect=1
IntensityAmount=0.1
IntensityEffect=2

[ColorEffects:Inactive]
ChangeSelectionColor=true
Color=162,157,163
ColorAmount=0.025
ColorEffect=2
ContrastAmount=0.1
ContrastEffect=2
Enable=false
IntensityAmount=0
IntensityEffect=0

[Colors:Button]
BackgroundAlternate=193,188,194
BackgroundNormal=255,252,255
DecorationFocus=31,197,152
DecorationHover=31,197,152
ForegroundActive=31,197,152
ForegroundInactive=132,126,133
ForegroundLink=83,166,225
ForegroundNegative=240,62,77
ForegroundNeutral=243,119,53
ForegroundNormal=40,38,41
ForegroundPositive=151,189,45
ForegroundVisited=191,101,240

[Colors:Complementary]
BackgroundAlternate=193,188,194
BackgroundNormal=255,252,255
DecorationFocus=31,197,152
DecorationHover=31,197,152
ForegroundActive=31,197,152
ForegroundInactive=132,126,133
ForegroundLink=83,166,225
ForegroundNegative=240,62,77
ForegroundNeutral=243,119,53
ForegroundNormal=40,38,41
ForegroundPositive=151,189,45
ForegroundVisited=132,126,133

[Colors:Header]
BackgroundAlternate=255,252,255
BackgroundNormal=255,252,255
DecorationFocus=31,197,152
DecorationHover=31,197,152
ForegroundActive=31,197,152
ForegroundInactive=132,126,133
ForegroundLink=83,166,225
ForegroundNegative=240,62,77
ForegroundNeutral=243,119,53
ForegroundNormal=40,38,41
ForegroundPositive=151,189,45
ForegroundVisited=191,101,240

[Colors:Header][Inactive]
BackgroundAlternate=255,252,255
BackgroundNormal=255,252,255
DecorationFocus=31,197,152
DecorationHover=31,197,152
ForegroundActive=31,197,152
ForegroundInactive=132,126,133
ForegroundLink=83,166,225
ForegroundNegative=240,62,77
ForegroundNeutral=243,119,53
ForegroundNormal=40,38,41
ForegroundPositive=151,189,45
ForegroundVisited=191,101,240

[Colors:Selection]
BackgroundAlternate=71,66,71
BackgroundNormal=238,186,33
DecorationFocus=31,197,152
DecorationHover=31,197,152
ForegroundActive=255,252,255
ForegroundInactive=193,188,194
ForegroundLink=242.25,202.5,88.5
ForegroundNegative=243.75,109.5,121.5
ForegroundNeutral=246,152.25,103.5
ForegroundNormal=255,252,255
ForegroundPositive=177,204.75,97.5
ForegroundVisited=207,138.75,243.75

[Colors:Tooltip]
BackgroundAlternate=255,252,255
BackgroundNormal=255,252,255
DecorationFocus=31,197,152
DecorationHover=31,197,152
ForegroundActive=31,197,152
ForegroundInactive=132,126,133
ForegroundLink=83,166,225
ForegroundNegative=240,62,77
ForegroundNeutral=243,119,53
ForegroundNormal=40,38,41
ForegroundPositive=151,189,45
ForegroundVisited=191,101,240

[Colors:View]
BackgroundAlternate=255,252,255
BackgroundNormal=255,252,255
DecorationFocus=31,197,152
DecorationHover=31,197,152
ForegroundActive=31,197,152
ForegroundInactive=132,126,133
ForegroundLink=83,166,225
ForegroundNegative=240,62,77
ForegroundNeutral=243,119,53
ForegroundNormal=40,38,41
ForegroundPositive=151,189,45
ForegroundVisited=191,101,240

[Colors:Window]
BackgroundAlternate=255,252,255
BackgroundNormal=255,252,255
DecorationFocus=31,197,152
DecorationHover=31,197,152
ForegroundActive=31,197,152
ForegroundInactive=132,126,133
ForegroundLink=83,166,225
ForegroundNegative=240,62,77
ForegroundNeutral=243,119,53
ForegroundNormal=40,38,41
ForegroundPositive=151,189,45
ForegroundVisited=191,101,240

[General]
ColorScheme=ThemerLight
Name=Themer Light
shadeSortColumn=true

[KDE]
contrast=4

[WM]
activeBackground=255,252,255
activeBlend=40,38,41
activeForeground=40,38,41
inactiveBackground=255,252,255
inactiveBlend=132,126,133
inactiveForeground=132,126,133"
`;
